relay-profiling/src/transaction_metadata.rs

```diff
@@ -30,6 +30,10 @@ impl TransactionMetadata {
             && self.relative_start_ns < self.relative_end_ns
             && self.relative_cpu_start_ms <= self.relative_cpu_end_ms
     }
+
+    pub fn duration_ns(&self) -> u64 {
+        self.relative_end_ns - self.relative_start_ns
```

Comment:
This could panic if `end < start`, you might want to use [saturating_sub](https://doc.rust-lang.org/std/primitive.u64.html?search=a#method.saturating_sub).